sphinx/texinputs/sphinx.sty

```diff
@@ -716,10 +716,38 @@
       \setbox\spx@image@box\box\voidb@x % clear memory
       \includegraphics[#1,width=\linewidth]{#2}%
     \else
+      \setbox\spx@image@box\box\voidb@x % clear memory
       \includegraphics[#1]{#2}%
     \fi
   \fi
 }
+% \sphinxsafeincludegraphics resizes images larger than the line width,
+% or taller than about the text height (whether or not height/width options
+% were used). This is requested to avoid a crash with \MakeFramed as used by
+% sphinxShadowBox (topic/contents) and sphinxheavybox (admonitions), and also
+% by sphinxVerbatim (but a priori no image inclusion there).
+\newdimen\spx@image@maxheight
+% default maximal setting will get reduced by sphinxShadowBox/sphinxheavybox
+\AtBeginDocument{\spx@image@maxheight\textheight}
```

Comment:
In this PR only sphinxShadowBox and sphinxheavybox will update value of `\spx@image@maxheight`. Also sphinxVerbatim uses framing: but image inclusion is a priori not possible there, so I did not add the corresponding code to it to update `\spx@image@maxheight`.

Reply:
Actually, an image could possibly be part of the caption of a code-block, and the caption is rendered using framing code, but I figured it was not needed to enter into unneeded complications as inclusion of a big image in such a caption title is really improbable.